src/puppetlabs/jdbc_util/core.clj

```diff
@@ -63,20 +64,30 @@
   ([result-set]
    (convert-result-arrays vec result-set))
   ([f result-set]
-   (let [convert #(cond
-                    (ks/array? %) (f %)
-                    (isa? (class %) java.sql.Array) (f (.getArray %))
-                    :else %)]
+   (let [convert (fn [v] (cond
+                           (ks/array? v) (f v)
+                           (instance? java.sql.Array v) (f (.getArray v))
+                           :else v))]
      (map #(ks/mapvals convert %) result-set))))
 
+(defn convert-result-pgobjects
+  "Converts PGObjects in a result set to be the value that they contain.
+  Values which aren't arrays are unchanged."
+  [result-set]
+  (let [val-if-pgobj (fn [v] (if (instance? PGobject v)
+                               (.getValue v)
+                               v))]
+    (map #(ks/mapvals val-if-pgobj %) result-set)))
+
 (defn query
   "An implementation of query that returns a fully evaluated result (no
-  JDBCArray objects, etc)"
+  lazy sequences, JDBCArray objects, or PGObjects)."
   [db sql-and-params]
   (let [convert (fn [rs]
                   (doall
-                    (convert-result-arrays vec
-                                           (jdbc/result-set-seq rs))))]
+                    (->> (jdbc/result-set-seq rs)
+                      (convert-result-arrays vec)
+                      (convert-result-pgobjects))))]
```

Comment:
Why not? 

RBAC was basically doing this before with that `extend-protocol` invocation, and that was way worse because there was no reliable way around it. This way, you're only subject to the behavior of this function if you opt in to using it. 

It's a pretty straightforward function, and most of its behavior is available piecemeal via the `convert-result-*` functions, so if you decide you don't like this function's behavior, then it's pretty easy to roll your own by just selecting the ones you want to apply.

My understanding is that everybody currently using this library wants this behavior, so it makes sense to add it here, as it functions as an easy way to get the postgres result transformations we all agree are useful. Is my understanding wrong?